scripts/healthcheck-api.sh

```
@@ -70,11 +81,67 @@ hc_ok() {
+# So the very first pm2 call must happen inside its own transient scope: if a
+# daemon needs spawning it lands in that separate cgroup and outlives this unit.
+# If one is already running, ping simply connects to it and the empty scope is
+# garbage-collected by --collect. Inspect with:
+#   systemctl --user list-units 'ac-api-pm2-*.scope'
+ensure_pm2_daemon() {
+  if [[ "$API_HEALTH_USE_SYSTEMD_RUN" != "1" ]] || ! command -v systemd-run >/dev/null 2>&1; then
+    echo "systemd-run unavailable; PM2 daemon will start in the caller's cgroup"
+    return 0
+  fi
+
+  local scope_name="ac-api-pm2-$$-$(date +%s)"
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)